src/utils/numAbbr.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @flow
+ * @file Function to abbreviate a number as a string in a locale-sensitive manner.
+ * @author Box
+ */
+
+import IntlMessageFormat from 'intl-messageformat';
+import { languages, numbers } from 'box-locale-data';
+
+interface NumAbbrOptions {
+    length: string;
+    locale: string;
+    numbersData: object;
+}
+
+function numAbbrWithLocale(numbersData: object, num: number, locale: string, options?: NumAbbrOptions): string {
+    if (!num) return '0';
+    let { length } = options || {};
+    length = length || 'short';
+    let exponent: number = Math.floor(num).toString().length - 1;
+    if (num < 0) {
+        exponent -= 1; // take care of the negative sign
+    }
+    const formats = numbersData[length];
+    const digits: number = exponent >= formats.length ? exponent - formats.length + 3 : formats[exponent].digits;
+    const count: number = Math.round(num / 10 ** (exponent - digits + 1));
+    const template = new IntlMessageFormat(
+        formats[exponent > formats.length ? formats.length - 1 : exponent].msg,
+        locale,
+    );
+    return template.format({ count });
+}
+
+/**
+ * Gets the number in abbreviated form in a locale-sensitive manner. This function
+ * scales the number down to the smallest it can be, taking only up to 4 significant
+ * digits, and rounding the rest. ie. 12345678 becomes "12M" in English.
+ *
+ * The abbreviation words/letters can have the length "short" or "long", specified
+ * with the "length" property in the options. If "short",
+ * then this function uses an abbreviation of the bucket such as "12M". If "long",
+ * then the name of the bucket is written out in full, such as "12 million".
+ *
+ * For locales that have complex plurals, such as Russian or Polish, this function
+ * returns the correctly pluralized suffix/prefix to go along with the scaled number.
+ *
+ * @param {*} num the number or numbers to abbreviate. (also accepts numbers in string form,
+ * or an array of numbers).
+ * @param {NumAbbrOptions=} options options governing how the output is generated
+ * @return {string} the number in abbreviated form as a string, or '0' if there was
+ * invalid input such as null instead of a number
+ */
+export default function(num: number, options?: NumAbbrOptions): string {
+    if (!num) return '0';
+
+    // languages contains info about the current locale
+    let { locale, numbersData } = options || {};
+    locale = locale || languages.bcp47Tag || 'en-US';
+    numbersData = numbersData || numbers;
+
+    switch (typeof num) {
+        case 'boolean':
+            num = num ? 1 : 0;
+            break;
+
+        case 'string':
+            num = parseInt(num, 10);
+            if (Number.isNaN(num)) return '0';
```

Comment:
Number.isNaN doesn't work in IE11. Probably use lodash/isNaN

Reply:
done